python/cudf/cudf/core/buffer/buffer.py

```diff
@@ -171,6 +270,112 @@ def __getitem__(self, key: slice) -> Buffer:
             raise ValueError("slice must be C-contiguous")
         return self._getitem(offset=start, size=stop - start)
 
+    def _is_cai_zero_copied(self):
+        """
+        Returns a flag, that indicates if the Buffer has been zero-copied.
+        """
+        return self._zero_copied
+
+    def _update_ref(self):
+        """
+        Generate the new proxy reference.
+        """
+        if (self._ptr, self._size) not in Buffer._refs:
+            Buffer._refs[(self._ptr, self._size)] = BufferWeakref(
+                self._ptr, self._size
+            )
+        self._proxy_ref = Buffer._refs[(self._ptr, self._size)]
```

Comment:
I added a todo here, since I want to be able to merge this into the Spillable buffer in a later PR and probably don't want to add multiple locks in our code.